### PR TITLE
Tweaks: Fixes "hide mini follow buttons" for new dash

### PR DESCRIPTION
**Pull request by xenticore**

### CODE REVIEW

File: Extensions/tweaks.js [hunk 1]
```diff
@@ -718,8 +718,15 @@ XKit.extensions.tweaks = new Object({
 			}
 		}
 
-		if (XKit.extensions.tweaks.preferences.hide_follows.value) {
-			XKit.extensions.tweaks.add_css(".note a.follow {display: none !important; } .notification_follow, .reblog_follow_button { display: none !important; } #posts .notes_outer_container.popover .note a.follow { display: none !important; } #posts .notes_outer_container.popover .note.like a.block { right: 16px; }", "xkit_tweaks_hide_follows");
```
Comment (nightpool) [inline Extensions/tweaks.js:722]: Would you mind leaving the old code in? it's still accessible in certain situations

File: Extensions/tweaks.js [hunk 2]
```diff
@@ -718,8 +718,15 @@ XKit.extensions.tweaks = new Object({
...
+		if (XKit.extensions.tweaks.preferences.hide_follows.value && XKit.page.react) {
+			let followButtonSelectors = XKit.css_map.keyToClasses('followButton');
+			let postSelectors = XKit.css_map.keyToClasses('post');
+			let postFollowButtonSelectors = postSelectors.map(postCssClass => {
+				return followButtonSelectors.map(followCssClass => `.${postCssClass} .${followCssClass}`).join(', ');
+			}).join(', ');
+			XKit.extensions.tweaks.add_css(`${postFollowButtonSelectors} {
+				display: none!important;
```
Comment (nightpool) [inline Extensions/tweaks.js:728]: nit: space between "none" and "!important"

File: Extensions/tweaks.js [hunk 3]
```diff
@@ -718,8 +718,15 @@ XKit.extensions.tweaks = new Object({
 			}
 		}
 
-		if (XKit.extensions.tweaks.preferences.hide_follows.value) {
-			XKit.extensions.tweaks.add_css(".note a.follow {display: none !important; } .notification_follow, .reblog_follow_button { display: none !important; } #posts .notes_outer_container.popover .note a.follow { display: none !important; } #posts .notes_outer_container.popover .note.like a.block { right: 16px; }", "xkit_tweaks_hide_follows");
+		if (XKit.extensions.tweaks.preferences.hide_follows.value && XKit.page.react) {
+			let followButtonSelectors = XKit.css_map.keyToClasses('followButton');
+			let postSelectors = XKit.css_map.keyToClasses('post');
+			let postFollowButtonSelectors = postSelectors.map(postCssClass => {
```
Comment (nightpool) [inline Extensions/tweaks.js:724]: flatMap might make this a little clearer (and I think the browser support is there? I haven't checked recently though)

File: Extensions/tweaks.js [hunk 4]
```diff
@@ -718,8 +718,15 @@ XKit.extensions.tweaks = new Object({
...
+				return followButtonSelectors.map(followCssClass => `.${postCssClass} .${followCssClass}`).join(', ');
```
Comment (nightpool) [inline Extensions/tweaks.js:725]: single line arrow functions can be braceless, and shouldn't have a return statement (I think eslint will also ding you for this)

File: Extensions/tweaks.js (hunk 3 above)
Comment (nightpool) [reply Extensions/tweaks.js:724]: Actually, I think there's a helper for this now? check xkit_patches, right under where keyToClasses is defined

**Comment by xenticore** [comment]
I've updated it, I wasn't sure how to keep compatibility so I copied code from pin_avatars.

File: Extensions/tweaks.js [hunk 5]
```diff
@@ -718,15 +718,15 @@ XKit.extensions.tweaks = new Object({
...
-			let postSelectors = XKit.css_map.keyToClasses('post');
-			let postFollowButtonSelectors = postSelectors.map(postCssClass => {
-				return followButtonSelectors.map(followCssClass => `.${postCssClass} .${followCssClass}`).join(', ');
-			}).join(', ');
-			XKit.extensions.tweaks.add_css(`${postFollowButtonSelectors} {
-				display: none!important;
-			}`, 'xkit_tweaks_hide_follows');
+		if (XKit.extensions.tweaks.preferences.hide_follows.value) {
+			if (XKit.page.react) {
+				let postFollowButtonSelectors = XKit.css_map.keyToCss('.post .followButton');
```
Comment (nightpool) [inline Extensions/tweaks.js:723]: close! this needs to be `XKit.css_map.descendantSelector(key1, key2)`

Comment (xenticore) [reply Extensions/tweaks.js:723]: `XKit.css_map.descendantSelector('post', 'followButton')` 
or 
`XKit.css_map.descendantSelector('.post', '.followButton')` ?

**Pull request merged**